powerauth-push-server/src/main/java/io/getlime/push/repository/InboxRepository.java

```diff
@@ -35,34 +35,38 @@ public interface InboxRepository extends PagingAndSortingRepository<InboxMessage
     /**
      * Find all messages for given user ID.
      * @param userId User ID.
+     * @param appId Application ID.
      * @param pageable Paging parameters.
      * @return List of messages for user ID.
      */
-    List<InboxMessageEntity> findAllByUserIdOrderByTimestampCreatedDesc(String userId, Pageable pageable);
+    List<InboxMessageEntity> findAllByUserIdAndAppIdOrderByTimestampCreatedDesc(String userId, String appId, Pageable pageable);
 
     /**
      * Find messages for given user ID in a given read state.
      * @param userId User ID.
+     * @param appId Application ID.
      * @param read Should the query return read messages, or those that are not read?
      * @param pageable Paging parameters.
      * @return List of messages for user ID with provided read state.
      */
-    List<InboxMessageEntity> findAllByUserIdAndReadOrderByTimestampCreatedDesc(String userId, boolean read, Pageable pageable);
+    List<InboxMessageEntity> findAllByUserIdAndAppIdAndReadOrderByTimestampCreatedDesc(String userId, String appId, boolean read, Pageable pageable);
 
     /**
      * Find first message with given ID and user ID.
-     * @param id Message ID.
+     * @param inboxId Message ID.
      * @param userId User ID.
+     * @param appId Application ID.
      * @return First message matching ID and user ID.
      */
-    Optional<InboxMessageEntity> findFirstByIdAndUserId(String id, String userId);
+    Optional<InboxMessageEntity> findFirstByInboxIdAndUserIdAndAppId(String inboxId, String userId, String appId);
```

Comment:
Just consider _parameter object_ pattern, three parameters in a row of the same type are error prone.

Reply:
Exactly.